rust/headless-client/src/ipc/linux.rs

```diff
@@ -11,30 +12,19 @@ pub(crate) type Stream = UnixStream;
 
 impl Server {
     /// Platform-specific setup
-    pub(crate) async fn new() -> Result<Self> {
-        Self::new_with_path(&crate::platform::sock_path()).await
-    }
-
-    /// Uses a test path instead of what prod uses
-    ///
-    /// The test path doesn't need admin powers and won't conflict with the prod
-    /// IPC service on a dev machine.
-    #[cfg(test)]
-    pub(crate) async fn new_for_test() -> Result<Self> {
-        let dir = crate::known_dirs::runtime().context("Can't find runtime dir")?;
-        // On a CI runner, the dir might not exist yet
-        tokio::fs::create_dir_all(&dir).await?;
-        let sock_path = dir.join("ipc_test.sock");
-        Self::new_with_path(&sock_path).await
-    }
-
-    async fn new_with_path(sock_path: &Path) -> Result<Self> {
```

Comment:
Is this not the better abstraction across all platforms? Just a ctor with a path and we treat that as a UDS or pipe depending on the platform?